CriticalMass/LocationManager.swift

```diff
@@ -24,75 +47,117 @@ class LocationManager: NSObject, CLLocationManagerDelegate, LocationProvider {
         }
     }
 
-    private var updateLocationCompletion: ResultCallback<Location>?
+    private var updateLocationCompletion: ResultCallback<CLLocation>?
+    private var locationRequests: [GeolocationCompletion] = []
+    private var authRequests: [GeolocationAuthCompletion] = []
+
+    private(set) var currentLocation: CLLocation?
 
-    func updateLocation(completion: ResultCallback<Location>?) {
-        updateLocationCompletion = completion
-        locationManager.requestLocation()
+    private let locationManager: CLLocationManager
+
+    init(
+        manager: CLLocationManager = CLLocationManager(),
+        locationProperties: LocationProperties = LocationProperties()
+    ) {
+        locationManager = manager
+        super.init()
+        configureLocationManager(with: locationProperties)
+        startTrackingLocation()
```

Comment:
Not sure if we still need. to call stratTrackingLocation on Init since the getCurrentLocation method is doing. that anyway